src/libclipper/test/policy_test.cpp

```diff
@@ -94,8 +98,21 @@ TEST_F(Exp3Test, UpdateTest) {
 
 TEST_F(Exp3Test, SelectionTest) {
   auto query = Utility::create_query(models);
-  auto tasks = Exp3Policy::select_predict_tasks(state, query, 1000);
-  ASSERT_NE(model_3.second, tasks.front().model_.second);
+  int select_1 = 0;
+  int select_2 = 0;
+  int select_3 = 0;
+  for (int i=0; i<100; ++i) {
+    auto tasks = Exp3Policy::select_predict_tasks(state, query, 1000);
+    if (model_1.second == tasks.front().model_.second) {
+      select_1 ++;
+    } else if (model_2.second == tasks.front().model_.second) {
+      select_2 ++;
+    } else {
+      select_3 ++;
+    };
+    ASSERT_GE(select_1, select_2);
+    ASSERT_GE(select_2, select_3);
```

Comment:
Does this test pass? I would expect that the asserts should be outside of the for loop?